pkg/edgeview/src/basics.go

```diff
@@ -0,0 +1,665 @@
+	}
+	return false
+}
+
+func evLogger(deb bool) *logrus.Logger {
+	formatter := logrus.JSONFormatter{
+		TimestampFormat: time.RFC3339Nano,
+	}
+	logger := logrus.New()
+	logger.SetFormatter(&formatter)
+	log = base.NewSourceLogObject(logger, agentName, os.Getpid())
+	if deb {
+		logger.SetLevel(logrus.DebugLevel)
+	}
+	return logger
+}
+
+func getBasics() {
+	if runOnServer {
+		if _, err := os.Stat("/config"); err != nil {
+			return
+		}
+	}
+
+	ips := []string{}
+	for _, i := range allUPIntfIPv4() {
+		if strings.HasPrefix(i.intfName, "bn") {
+			continue
+		}
+		ips = append(ips, i.ipAddr)
+	}
+	if myEvEndPoint != "" {
+		fmt.Printf("Device IPs: %v; Endpoint IP %s\n", ips, myEvEndPoint)
+	} else {
+		fmt.Printf("Device IPs: %v\n", ips)
+	}
+
+	retbytes, err := ioutil.ReadFile("/persist/status/uuid")
+	if err == nil {
+		fmt.Printf("  UUID: %s", string(retbytes))
+	}
+	retbytes, err = ioutil.ReadFile("/config/server")
+	if err == nil {
+		var printed bool
+		serverurl := string(retbytes)
+		conts := strings.Split(serverurl, "zedcloud.")
+		if len(conts) == 2 {
+			cont2s := strings.Split(conts[1], ".zededa.net")
+			if len(cont2s) == 2 {
+				cluster := cont2s[0]
+				controller := strings.Replace(serverurl, cluster, "\033[0;93m"+cluster+"\033[0m", 1)
+				fmt.Printf("  Controller: %s", controller)
+				printed = true
+			}
+		}
+		if !printed {
+			fmt.Printf("  Controller: %s", serverurl)
+		}
+	}
+
+	var release, partition string
+	retbytes, err = ioutil.ReadFile("/run/eve-release")
+	if err == nil {
+		release = string(retbytes)
+	}
+	retbytes, err = ioutil.ReadFile("/run/eve.id")
+	if err == nil {
+		partition = string(retbytes)
+	}
+
+	fmt.Printf("  EVE-OS release %s, %s", release, partition)
+	fmt.Printf("  Edge-View Ver: %s, JWT expires at %v\n",
+		edgeViewVersion, time.Unix(int64(evStatus.ExpireOn), 0).Format(time.RFC3339))
+	hinfo, err := host.Info()
+	if err == nil {
+		loc, _ := time.LoadLocation("UTC")
+		fmt.Printf("  %v(%v), uptime %d (sec) = %d days",
+			time.Now().In(loc).Format(time.RFC3339), loc, hinfo.Uptime, hinfo.Uptime/(3600*24))
+	}
+	fmt.Println()
+	fmt.Println()
+	closePipe(true)
+}
+
+// dynamically install package for uncommon options
+func addPackage(pathName, pkgName string) error {
+	_, err := os.Stat(pathName)
+	if err != nil {
+		if os.IsNotExist(err) {
+			_, err = runCmd("apk add "+pkgName, false)
+			if err != nil {
+				fmt.Printf("%v\n", err)
+				return err
+			}
+		} else {
+			return err
+		}
+	}
+	return nil
+}
+
+func runCmd(cmd string, isPrint bool) (string, error) {
+	var retStr string
+	var retBytes []byte
+	var err error
+
+	retBytes, err = runCmdInFunction(cmd)
+	if err != nil {
+		if !strings.HasSuffix(err.Error(), "status 1") {
+			fmt.Printf("error: %v\n", err)
+		}
+	} else {
+		retStr = string(retBytes)
+		if isPrint {
+			fmt.Println(retStr)
+			closePipe(true)
+		}
+	}
+	return retStr, err
+}
+
+func runCmdInFunction(cmdOpt string) ([]byte, error) {
+	shellcmd := strings.Fields(cmdOpt)
```

Comment:
Nit on variables names, but since there is no shell any more it would make sense to rename the 30-odd occurances of shell and shellcmd to just cmd.

Reply:
will fix them all.